### PR TITLE
Add cuda.device_count api

### PR DESCRIPTION
**Pull request by linjieccc**
### PR types

New features
### PR changes

APIs
### Describe

Add paddle.device.cuda.device_count API to return the number of available GPUs.

image

image

### CODE REVIEW

File: python/paddle/device/cuda/__init__.py [hunk 1]
```diff
@@ -94,3 +95,25 @@ def synchronize(device=None):
             raise ValueError("device type must be int or paddle.CUDAPlace")
 
     return core._device_synchronize(device_id)
+
+
+def device_count():
+    '''
+    Return the number of GPUs available.
+    
+    Returns:
```
Comment (wawltor) [inline python/paddle/device/cuda/__init__.py:104]: 看起来没有明确return type

Comment (linjieccc) [reply python/paddle/device/cuda/__init__.py:104]: 已修改

**Review by TCChenlong — APPROVED** [review]

**Review by wawltor — APPROVED** [review]
LGTM

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**